python/tank/context.py

```diff
@@ -761,12 +754,73 @@ def deserialize(cls, context_str):
 
         # create a Sgtk API instance.
         tk = Tank(pipeline_config_path)
+        data["tk"] = tk
 
         # add it to the constructor instance
+        # and lastly make the obejct
+        return cls._from_dict(data)
+
+    def to_dict(self):
+        """
+        Converts the context into a dictionary with keys ``project``,
```

Comment:
I've chosen to not repeat the mistake of serializing the user inside the dictionary, as this has been error prone for clients submitting jobs on their farm where they serialize the context. This pattern feels right.